doc/manual/methods.rst

```diff
@@ -552,4 +558,164 @@ arguments::
     function emptyfunc
     end
 
+.. _man-method-design-ambiguities:
+
+Method design and the avoidance of ambiguities
+----------------------------------------------
+
+Julia's method polymorphism is one of its most powerful features, yet
+exploiting this power can pose design challenges.  In particular, in
+more complex method hierarchies it is not uncommon for
+:ref:`ambiguities <man-ambiguities>` to arise.
+
+Above, it was pointed out that one can resolve ambiguities like
+::
+
+   f(x, y::Int) = 1
+   f(x::Int, y) = 2
+
+by defining a method
+::
+
+   f(x::Int, y::Int) = 3
+
+This is often the right strategy; however, there are circumstances
+where following this advice blindly can be counterproductive. In
+particular, the more methods you have, the more possibilities there
```

Comment:
Just to be on the save side, I'd write "...,the more methods a generic function has,..."